package.json

```diff
@@ -1,34 +1,55 @@
 {
-  "name": "@oasisprotocol/deoxysii",
-  "version": "0.0.5",
-  "description": "Deoxys-II-256-128",
-  "main": "./deoxysii",
-  "types": "./deoxysii.d.ts",
-  "scripts": {
-    "build": "tsc",
-    "bench": "node ./bench/bench.js",
-    "test": "mocha",
-    "lint": "eslint *.js ./test ./bench"
-  },
-  "repository": {
-    "type": "git",
-    "url": "https://github.com/oasisprotocol/deoxysii-js.git"
-  },
-  "license": "MIT",
-  "author": "Oasis Protocol Foundation <info@oasisprotocol.org>",
-  "devDependencies": {
-    "@types/chai": "^4.3.4",
-    "@types/mocha": "^10.0.1",
-    "@types/node": "^18.15.11",
-    "@types/uint32": "^0.2.0",
-    "benchmark": "^2.1.4",
-    "chai": "*",
-    "eslint": "^5.14.0",
-    "mocha": "^6.1.4",
-    "typescript": "^5.0.4"
-  },
-  "dependencies": {
-    "bsaes": "0.0.2",
-    "uint32": "^0.2.1"
-  }
+	"name": "@oasisprotocol/deoxysii",
+	"version": "0.0.6",
+	"description": "Deoxys-II-256-128",
+	"sideEffects": false,
+	"files": ["dist", "!dist/*.tsbuildinfo", "src", "!src/*.json"],
+	"main": "./dist/_cjs/deoxysii.js",
+	"module": "./dist/_esm/deoxysii.js",
+	"types": "./dist/_types/deoxysii.d.ts",
+	"typings": "./dist/_types/deoxysii.d.ts",
+	"exports": {
+		"node": {
+			"import": "./dist/_esm/deoxysii.js",
+			"require": "./dist/_cjs/deoxysii.cjs",
+			"types": "./dist/_types/deoxysii.d.ts"
+		},
+		"types": "./dist/_types/deoxysii.d.ts",
+		"default": "./dist/_esm/deoxysii.js"
+	},
+	"scripts": {
+		"bench": "node ./bench/bench.js",
+		"test": "vitest",
+		"coverage": "vitest run --coverage --coverage.exclude=scripts --coverage.exclude='src/*.bench.ts'",
+		"lint": "biome check .",
+		"format": "biome format --write .",
+		"clean": "rm -rf dist",
+		"build": "npm run build:cjs && npm run build:esm && npm run build:types",
+		"build:cjs": "tsc --project ./tsconfig.build.json --module commonjs --outDir ./dist/_cjs --removeComments --verbatimModuleSyntax false && printf '{\"type\":\"commonjs\"}' > ./dist/_cjs/package.json && node scripts/rename-cjs.mjs",
+		"build:esm": "tsc --project ./tsconfig.build.json --module es2015 --outDir ./dist/_esm && printf '{\"type\": \"module\",\"sideEffects\":false}' > ./dist/_esm/package.json",
+		"build:types": "tsc --project ./tsconfig.build.json --module esnext --declarationDir ./dist/_types --emitDeclarationOnly --declaration --declarationMap",
+		"prepublishOnly": "npm run build"
+	},
+	"repository": {
+		"type": "git",
+		"url": "https://github.com/oasisprotocol/deoxysii-js.git"
+	},
+	"license": "MIT",
```

Comment:
deoxysii.ts & aes.ts are still MIT licensed

The uint32 package doesn't really have a license (and is explicitly noted by the author as something you can do whatever tf you want with it), so that's been noted in an appropriate `SPDX-License-Identifier`. I believe it would be entirely kosher to strip the license header, add my own copyright and re-license it as MIT if I wanted.

Reply:
o ok

Reply:
> strip the license header

sounds like that's allowed

> add my own copyright

I don't think the license controls whether you can do that, and I think you can't do that

> re-license it as MIT

I think you have enough permission to redistribute it under MIT, which is more restrictive

Reply:
It would be fair to add a copyright notice to it given that the changes I made to it are, while trivial, about as much effort as writing it in the first place.

Either way I'm not fussed, please advise on what to do with `uint32.ts`

Reply:
if it was done by changing the uint32.js albeit extensively, that sounds like it's a derivative work. I've read that it's usually preferred to contribute to a project under its original license, which for the uint32.js library was a custom license. so overall, this deoxysii-js package would have a custom license, of what would be expressed in pseudo-SPDX as (MIT AND uint32.js's license)